src/test/chainstate_write_tests.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2024- The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/setup_common.h>
+#include <validation.h>
+#include <validationinterface.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(chainstate_write_tests)
+
+struct TestSubscriber final : public CValidationInterface {
+    bool m_did_flush{false};
+
+    TestSubscriber() = default;
+
+    void ChainStateFlushed(ChainstateRole role, const CBlockLocator& locator) override
+    {
+        m_did_flush = true;
+    }
+};
+
+BOOST_FIXTURE_TEST_CASE(chainstate_write_interval, TestingSetup)
```

Comment:
Besides the stability and the initial run, I think we should test the valid values as well inside the range, e.g.
```C++
BOOST_FIXTURE_TEST_CASE(chainstate_write_interval, TestingSetup)
{
    auto DATABASE_WRITE_INTERVAL_MIN{50min};
    auto DATABASE_WRITE_INTERVAL_MAX{70min};
    auto now = GetTime<std::chrono::minutes>();
    auto sub{std::make_shared<TestSubscriber>()};
    m_node.validation_signals->RegisterSharedValidationInterface(sub);
    auto test_flush = [&](std::chrono::minutes offset) {
        assert(!sub->m_did_flush);
        Chainstate& chainstate{Assert(m_node.chainman)->ActiveChainstate()};
        BlockValidationState state_dummy{};

        SetMockTime(now + offset);
        chainstate.FlushStateToDisk(state_dummy, FlushStateMode::PERIODIC);
        m_node.validation_signals->SyncWithValidationInterfaceQueue();
        return sub->m_did_flush;
    };

    // Initial run shouldn't flush
    BOOST_CHECK_EQUAL(test_flush(DATABASE_WRITE_INTERVAL_MIN + 5min), false);

    // Test all values (+1) in the interval
    BOOST_CHECK_EQUAL(test_flush(DATABASE_WRITE_INTERVAL_MIN - 1min), false);
    for (auto m = DATABASE_WRITE_INTERVAL_MIN; m <= DATABASE_WRITE_INTERVAL_MAX + 1min; m += 1min) {
        BOOST_CHECK_EQUAL(test_flush(m), true);
    }
}
```

Reply:
Why would we need to test every  value in the window? Testing the edges of the window accomplishes the same thing, no?

Reply:
Mostly because I ran it and it behaved in a weird way (sometimes passing) so I have a hard time trusting it now. It never passed for ALL of these values, so at least they're more reliable.

Reply:
Using SteadyClock in https://github.com/bitcoin/bitcoin/pull/30611#discussion_r1741105265 seems to have stabilized the tests.
Approach ACK